interface.py

```diff
@@ -572,51 +574,63 @@ def upload():
 
     files = request.files.getlist("file[]")
     if not files:
-        return redirect("./", code=400)
+        abort(400)
+
+    response_code = 303
```

Comment:
I think returning the `response_code` after processing the list is on the right track, but I see two problems with it still:

1. The first one is that you could override the response code more than once. Let's say you try to upload two files. First one has `''` for name, and second one is in some unsupported format (let's say it's `b.alac`). You would see the first error and make `response_code` be `400`, then see the second and make `response_code` be `415`. The code sent to the client would be `415` in the end. But if the order of files was inverted, the code sent would be `400`. Yet the action attempted by the client was exactly the same! This interaction means that not only would the response depend on the processing order; it would also omit some error information that could be useful to the client (by for example not sending the `400` code for the first file's error because the second file's error overwrote it). 
2. The second one is that you could be returning a `4xx` code and at the same time having a side-effect on the server. Like, if I receive a `4xx` code I would assume that no permanent action was taken by the server. I think this is also true for `5xx` codes, but I'm not sure.

I've consulted this with some friends who are more adept at http than I am, and they suggested something like this:
* Make a dictionary with the response for each file upload
* Send that dictionary to the client, together with a `200 OK` code or similar.

I think it's an interesting idea. 
Maybe instead of `200 OK` we'd want to keep using the current `303 See Other` to redirect the user to the main page.

Reply:
Processing the files one at a time could avoid this issue.

However, I do like the idea of returning an overall response (possibly just a 400) and returning data for the individual file upload success/fail statuses.